File: cserver/src/main/java/com/example/coffeeshops/coffee/CoffeeShop.java
package com.example.coffeeshops.coffee;

import jakarta.persistence.*;

@Entity
@Table(name = "coffee_shop")
public class CoffeeShop {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String address;
    private double lat;
    private double lng;
    private double rating;

    public CoffeeShop() {}

    public CoffeeShop(String name, String address, double lat, double lng, double rating) {
        this.name = name;
        this.address = address;
        this.lat = lat;
        this.lng = lng;
        this.rating = rating;
    }

    // ✅ Геттеры и сеттеры
    public Long getId() { return id; }
    public String getName() { return name; }
    public String getAddress() { return address; }
    public double getLat() { return lat; }
    public double getLng() { return lng; }
    public double getRating() { return rating; }

    public void setId(Long id) { this.id = id; }
    public void setName(String name) { this.name = name; }
    public void setAddress(String address) { this.address = address; }
    public void setLat(double lat) { this.lat = lat; }
    public void setLng(double lng) { this.lng = lng; }
    public void setRating(double rating) { this.rating = rating; }
}

File: cfront/lib/models/coffee_shop.dart
class CoffeeShop {
  final int id;
  final String name;
  final String address;
  final double lat;
  final double lng;
  final double rating;

  CoffeeShop({
    required this.id,
    required this.name,
    required this.address,
    required this.lat,
    required this.lng,
    required this.rating,
  });

  factory CoffeeShop.fromMap(Map<String, dynamic> map) => CoffeeShop(
        id: map['id'] is int ? map['id'] : int.tryParse(map['id'].toString()) ?? 0,
        name: map['name'] ?? '',
        address: map['address'] ?? '',
        lat: (map['lat'] ?? 0).toDouble(),
        lng: (map['lng'] ?? 0).toDouble(),
        rating: (map['rating'] ?? 0).toDouble(),
      );

  Map<String, dynamic> toMap() => {
        'id': id,
        'name': name,
        'address': address,
        'lat': lat,
        'lng': lng,
        'rating': rating,
      };

  @override
  String toString() =>
      '$name\n$address\nРейтинг: ${rating.toStringAsFixed(1)}';
}
